src/extensions/extension-discovery.ts

```diff
@@ -156,9 +156,18 @@ export class ExtensionDiscovery {
   }
 
   handleWatchFileAdd =  async (filePath: string) => {
-    if (path.basename(filePath) === manifestFilename) {
+    // e.g. "foo/package.json"
+    const relativePath = path.relative(this.localFolderPath, filePath);
+
+    // Converts "foo/package.json" to ["foo", "package.json"], where length of 2 implies
+    // that the added file is in a folder under local folder path.
+    // This safeguards against a file watch being triggered under a sub-directory which is not an extension.
+    const isUnderLocalFolderPath = relativePath.split(path.sep).length === 2;
```

Comment:
are you positive that this is correct. Windows does accept `/` as a path separator even though the common one is `\`

Reply:
I suspect that the paths from chokidar are using `\`, at least it's that way in the Windows CI pipeline. It would seem unusual to get `/` from that in some cases. I think `/` would be encountered only on some user input? But we're receiving file add events here.

Reply:
Fair, that is probably good enough.